### ISSUE TITLE
Cannot read properties of undefined (reading 'removeChildren')

### ISSUE THREAD

**Issue by Fallayn** [dev7355608/perfect-vision#225]
Sorry for the not particularly helpful issue title.

Perfect vision started suddenly throwing errors at me, canvas remains black.

Monk's appears in the log and I tested disabling it, but that makes no difference. Disabling just Perfect Vision fixes the issue.

```
libWrapper-wrapper.js:517 Uncaught TypeError: Cannot read properties of undefined (reading 'removeChildren')
[Detected 2 packages: perfect-vision, lib-wrapper]
    at SightLayer.initializeSources (foundry.js:30216:19)
    at Wrapper.🎁call_wrapped [as call_wrapped] (libWrapper-wrapper.js:511:22)
    at SightLayer.🎁SightLayer.prototype.initializeSources#perfect-vision (sight.js:59:16)
    at SightLayer.🎁SightLayer.prototype.initializeSources#0 (libWrapper-wrapper.js:187:52)
    at PerceptionManager._update (foundry.js:33290:44)
    at foundry.js:33272:54
initializeSources @ foundry.js:30216
🎁call_wrapped @ libWrapper-wrapper.js:511
🎁SightLayer.prototype.initializeSources#perfect-vision @ sight.js:59
🎁SightLayer.prototype.initializeSources#0 @ libWrapper-wrapper.js:187
_update @ foundry.js:33290
(anonym) @ foundry.js:33272
setTimeout (async)
_update @ foundry.js:33272
schedule @ foundry.js:33199
_refreshPerception @ foundry.js:37180
refresh @ foundry.js:36806
🎁call_wrapped @ libWrapper-wrapper.js:511
_levelsTileRefresh @ helpers.js:26
🎁Tile.prototype.refresh#perfect-vision @ tiles.js:44
(anonym) @ patch.js:52
(anonym) @ levels.js:67
wrapper @ patch.js:52
tileRefresh @ monks-active-tiles.js:4278
🎁Tile.prototype.refresh#0 @ libWrapper-wrapper.js:187
draw @ foundry.js:36713
await in draw (async)
🎁call_wrapped @ libWrapper-wrapper.js:511
🎁Tile.prototype.draw#perfect-vision @ tiles.js:9
🎁Tile.prototype.draw#0 @ libWrapper-wrapper.js:187
(anonym) @ foundry.js:22211
draw @ foundry.js:22209
await in draw (async)
draw @ foundry.js:29453
draw @ foundry.js:29723
🎁call_wrapped @ libWrapper-wrapper.js:511
🎁ForegroundLayer.prototype.draw#perfect-vision @ foreground.js:14
(anonym) @ patch.js:52
(anonym) @ levels.js:35
wrapper @ patch.js:52
🎁ForegroundLayer.prototype.draw#0 @ libWrapper-wrapper.js:187
draw @ foundry.js:20066
await in draw (async)
🎁call_wrapped @ libWrapper-wrapper.js:511
🎁Canvas.prototype.draw#perfect-vision @ canvas.js:403
🎁Canvas.prototype.draw#0 @ libWrapper-wrapper.js:187
view @ foundry.js:17809
_onClickScene @ foundry.js:53066
dispatch @ jquery.min.js:2
v.handle @ jquery.min.js:2
foundry.js:30216 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'removeChildren')
[Detected 2 packages: perfect-vision, lib-wrapper]
    at SightLayer.initializeSources (foundry.js:30216:19)
    at Wrapper.🎁call_wrapped [as call_wrapped] (libWrapper-wrapper.js:511:22)
    at SightLayer.🎁SightLayer.prototype.initializeSources#perfect-vision (sight.js:59:16)
    at SightLayer.🎁SightLayer.prototype.initializeSources#0 (libWrapper-wrapper.js:187:52)
    at PerceptionManager._update (foundry.js:33290:44)
    at PerceptionManager.update (foundry.js:33209:10)
    at PerceptionManager.initialize (foundry.js:33218:17)
    at Canvas._initialize (foundry.js:20227:21)
    at async Canvas.draw (foundry.js:20084:7)
    at async Canvas.🎁Canvas.prototype.draw#perfect-vision (canvas.js:403:9)
    at async Scene.view (foundry.js:17809:7)
```

**Comment by dev7355608** [comment]
Thanks for the report. This was already fixed in 3.7.3.

**Comment by Fallayn** [comment]
Ah, my bad, I made sure to hit update - maybe Forge caching. Happy to know it was fixed, and thanks for the awesome module!
